### DISCUSSION TITLE
Android improvements

### DISCUSSION THREAD

**Discussion by AnderYo** [ihhub/fheroes2#6490] [Ideas]
First of all, thank you very much for bringing this to life and to Android ... I've been playing it obsesively since I saw the article on Droid Gamers. Funny I started with H3 and never gave H2 a chance, but now that I gave it a shot - it has so much charm, it drew me in more than H3 has in a long while. 

After playing extensively on Android, I wanted to point out a few things that I ran into and I think would be great to improve on ... maybe some have been considered already but I couldn't find it searching through the discussions:
- Delete save file ... maybe a delete save button could be added or something along the lines? After long playing sessions I end up with tens of saves and it is a pain to scroll through them. Only way I found to delete files so far is to connect an external bluetooth and hit the "del" key. Weirdly enough I could not find them if searching through data, but Android has done so many weird things with it's storage in the latest versions ...
- Sleep button for heroes... I could not find if this option exists (I understand a shortcut exists for keyboard).
- Two click attack (first click selects, second confirms) or some other way to be able to attack and position your creature better ... Since on Android you basically press for attack, it is a bit difficult sometimes to see if you are pressing the sprite right so that your creature goes to the right place.
- Damage info ... it appears right where your finger is, which sometimes is a bit difficult to see (since you can't just hover). Maybe a solution here would be to also have that info on the lower part of the screen, for example in the lower bar where it says "Attack bone dragons".
- Random map generation ... I noticed someone already proposed this, I just wanted to add my voice to this. It would be so cool to be able to generate the random map on Android (I know you can do it on the PC through the basic H2 instalation and transfer it). I understand the map generation is a bit basic, not sure if it can also be improved more if added ...
- Keyboard on screen - I saw this was on the list of improvements already.
- Long press for right click. This is not that important I think, two fingers holding for right click works also (you get used to it), but long press seems more intuitive.

Some side notes: 
- If it helps, from a technical point of view, I've been playing it on my samsung fold 3 and the game has been super smooth and stable for me ... 
- Have played the good, the evil and POL campaign so far, the game has been crisp, smooth, felt great playing!
The only thing weird to me in the POL campaign and not sure if this was in the original as well, but the red guy in the last scenario  had like 1600 bone dragons, 1k something vampire lords and 900 liches when i reached him in month 4 (my math didn't add up on how he got to those numbers with 5 castles, so I guess he got some help from the scenario ;) ) ... I needed to cheese him a lot to win, which was a bit anticlimatic ...

I'll stop rambling now ... This h2 "ressurection" has been clearly done with a lot of love, so thank you all again for your work - amazing job!

**Comment by oleg-derevenetz** [comment]
> Long press for right click. This is not that important I think, two fingers holding for right click works also (you get used to it), but long press seems more intuitive.

Two fingers for the right click were selected deliberately because you will be able to just remove one of the fingers from the screen and see an info window under it (because info windows often appear too close to the mouse cursor), while with long press you will not be able to do this.

**Reply by AnderYo** [reply]
Ahh, I see what you mean. Indeed with long press you would need the window to pop up a bit farther from the finger ...
I think the two finger approach works fine as well, it just took a bit of getting used to at the beggining but afterwards it's very good as well.
